### PR TITLE
Expanded Playback and List controls

### LINKED ISSUE #2902: Separate control for autoplay next in list
To autoplay videos when you arrive on a page should be a separate control from autonavigating to the next in the list.

### LINKED ISSUE #6210: Ability to create (copy) playlist from Favorites / Watch later (or anyone's playlist)
We discussed this as a nice to have in the future - I can click an option to start a new playlist from my watch later/favorite/own or others' playlist.

### LINKED ISSUE #6220: Figure out how to show playlists in sidebar
https://github.com/lbryio/lbry-desktop/issues/6211#issuecomment-859492660

### LINKED ISSUE #6321: Countdown on lists
Is the countdown here needed?

Youtube only has it for recommended content

### PR DESCRIPTION
**Pull request by saltrafael**
🟢 Available for testing on https://salt.odysee.com 

## What is the new behavior?

- Don't show countdown on a list (Closes #6321)
- Added "Replay" option on autoplay (#2921)
image
- Added Loop option on Lists
- Added Shuffle option on Lists
- Added link to "View List" popup option, so can be opened on a new tab
- Clicking on the title of a floating player will take you back to the list being played
- Play Next/Previous (with shortcuts SHIFT+N SHIFT+P):
Peek 2021-08-20 12-05
- Separate control for autoplay next (Closes #2902)
Peek 2021-09-02 08-28
image

## PR Type & Checklist

<details><summary>PR Type...</summary>

What kind of change does this PR introduce?

- [X] Bugfix
- [X] Feature
- [X] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

</details>

<details><summary>PR Checklist...</summary>

Please check all that apply to this PR using "x":

- [X] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [ ] I added a line describing my change to CHANGELOG.md
- [ ] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below
</details>

this is a long one
🚀 the best is yet to come

### CODE REVIEW

File: ui/redux/actions/content.js [hunk 1]
```diff
@@ -298,17 +298,25 @@ export function doToggleLoopList(collectionId: string, loop: boolean, hideToast:
...
 
-export function doToggleShuffleList(collectionId: string, shuffle: boolean, hideToast: boolean) {
+export function doToggleShuffleList(currentUri: string, collectionId: string, shuffle: boolean, hideToast: boolean) {
   return (dispatch: Dispatch, getState: () => any) => {
     if (shuffle) {
       const state = getState();
       const urls = makeSelectUrlsForCollectionId(collectionId)(state);
 
-      const newUrls = urls
+      let newUrls = urls
```
Comment (jessopb) [inline ui/redux/actions/content.js:316]: Is it worth filtering out at least the top 5 in content/history reducer? Or even the top <collection_size>?

**Comment by jessopb** [comment]
This is huge.
I think copy list is its own thing.
modalCollectionAdd could conditionally return a simple collectionCopy component instead of the collectionAdd with the checkboxes etc.

**Comment by jessopb** [comment]
Not sure about adding lists to sidebare *in this way*. Will discuss at planning.

**Comment by jessopb** [comment]
It seems like coming up with a a shuffled next url can be done in the index.js of the a component, especially interacting and excluding based on content/history reducer. I'm not sure you need any work in lbry-redux for this.

**Comment by jessopb** [comment]
I don't quite understand the case for selecting the previous url from collection *with shuffle*.

**Comment by jessopb** [comment]
At the meeting we thought we needed to brainstorm the design for sidebar playlists more - take it out for now?

**Comment by saltrafael** [comment]
✔️ Separated Copy List
✔️ Took out Sidebar Lists

> It seems like coming up with a a shuffled next url can be done in the index.js of the a component, especially interacting and excluding based on content/history reducer. I'm not sure you need any work in lbry-redux for this.

IMO on redux it avoids repetition of the same code every time you want to select a shuffled list

> I don't quite understand the case for selecting the previous url from collection _with shuffle_.

going back on previously played shuffled videos, or else the ordering would be lost

**Comment by jessopb** [comment]
Ah, I see.
How about just selecting the values right in the selector like is done here:
https://github.com/lbryio/lbry-redux/blob/d016d8057be1c676db4ad0d21bdbcacd39a678eb/src/redux/selectors/publish.js#L43
state => state.content.shuffledUris,
state => state.content.shouldLoop,

**Comment by saltrafael** [comment]
> How about just selecting the values right in the selector like is done here:

✔️  Done, and wow this does look way cleaner
➕ Also just added "separate control for autoplay next"

**Comment by saltrafael** [comment]
> * [ ]  if autoplay is off in settings and you're in playlist mode, we don't show the autoplay next button - should this assume autoplay next is on, or should we just always show the button (I know youtube hide it).
> * [ ]  the settings option will only control autoplay on load (adjust text), and autoplay next will be controlled from the player. We can consider adding it settings also.

These are up for discussion

> * [ ]  if you have autoplay off + playing a video + then go to settings to enable autoplay, it crashes at :         autoplayButton.removeClass('vjs-button--autoplay-next');
>   ```
>     at videojs.jsx:670
>     at oc (react-dom.production.min.js:211)```  
>   ```

This is the last one missing, I didn't manage to reproduce it @tzarebczan

**Pull request merged**